### DISCUSSION TITLE
Automerge list of packages

### DISCUSSION THREAD

**Discussion by antondanilov95** [renovatebot/renovate#12597] [Request Help] [answered]
Hello, how I can renew only my list of libs?
I have the array of my libs and want to renew them with automerge, another libs not's interested me.
my config looks like this
`  packageRules: [
    {
      automerge: true,
    },
    {
      excludePackageNames: MY_DEPS_ARRAY,
      enabled: false,
      automerge: false,
    },
  ],`
but in this case, renovate make merge request with all libs in my package.json
Also i tried something like this
`
 "packageRules": [
    {
      enabled: false,
    {
      "matchPackageNames": MY_DEPS_ARRAY,
       automerge: true,
       enabled: true
    }
  ]
`
but this also not work as I'm expected

**Comment by rarkins** [comment, marked as answer]
You have invalid syntax in both the configs you mentioned, so I wouldn't expect either to work. Please don't paste what your config "looks like" and instead paste what it actually is. Or create a reproduction repo to demonstrate it publicly

**Reply by antondanilov95** [reply]
```

module.exports = {
  platform: 'gitlab',
  endpoint: process.env.CI_API_V4_URL,
  baseBranches: ['main', 'master'],
  npmrc: NPM_RC,
  includePaths: ["*"],
  onboardingConfig: {
    extends: ['config:base'],
  },
  repositories: REPOS_ARRAY,
  packageRules: [
    {
      automerge: true,
    },
    {
      excludePackageNames: MY_DEPS_ARRAY,
      enabled: false,
      automerge: false,
    },
  ],
}
```

**Reply by antondanilov95** [reply]
@rarkins full config ^

**Reply by rarkins** [reply]
Try:

```
  automerge: true,
  packageRules: [
    {
      excludePackageNames: MY_DEPS_ARRAY,
      enabled: false,
    },
  ],
```

But setting `automerge:true` is not recommended because it will apply to major updates too. Better:

```
  packageRules: [
    {
      matchUpdateTypes: ['digest', 'pin', 'patch', 'minor'],
      automerge: true,
    },
    {
      excludePackageNames: MY_DEPS_ARRAY,
      enabled: false
    },
  ],
```

**Reply by antondanilov95** [reply]
it works with my list of libs, but also create the second merge request which contains change versions of other libs in my projects

**Reply by rarkins** [reply]
Can't help further without a public reproduction repo

**Comment by rajadileepkolli** [comment]
Hi @rarkins , I am also facing the same issue. In this [PR](https://github.com/rajadileepkolli/POC/pull/403) I can see that AutoMerge is enabled but it is not getting merged. Can you please help me.

**Reply by viceice** [reply]
Check the logs
![image](https://user-images.githubusercontent.com/1798109/142578207-f2d72de0-e3c1-45d7-b68f-721e7b87bbf7.png)

The link is in PR footer
